helix-term/src/application.rs

```diff
@@ -189,14 +189,21 @@ impl Application {
                             Some(Layout::Horizontal) => Action::HorizontalSplit,
                             None => Action::Load,
                         };
+                        let old_id = editor.document_id_by_path(&file);
                         let doc_id = match editor.open(&file, action) {
                             // Ignore irregular files during application init.
                             Err(DocumentOpenError::IrregularFile) => {
                                 nr_of_files -= 1;
                                 continue;
                             }
                             Err(err) => return Err(anyhow::anyhow!(err)),
-                            Ok(doc_id) => doc_id,
+                            Ok(doc_id) => {
+                                // We can't open more than 1 buffer for 1 file, in this case we already have opened this file previously
+                                if old_id == Some(doc_id) {
```

Comment:
would be cleaner as an `if guard on the match arm

Reply:
Ok, do you mean e.g. this:

```rs
                            Ok(doc_id) if old_id == Some(doc_id) => {
                                nr_of_files -= 1;
                                doc_id
                            }
                            Ok(doc_id) => doc_id,
```

I did it in https://github.com/helix-editor/helix/pull/12199/commits/2558b3eb7f783fdb0fa981be3537c54d5671fb63